### PR TITLE
X-entropy assert probabilties sum to approximately one

### PR DESCRIPTION
**Pull request by beyarkay**
Currently, it's easy enough to forget that `from_logits` is false by default and it's easy to pass in logits when cross-entropy is expecting probabilities. This won't raise an error, because the values are scaled by [this line](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/backend.py#L4891).

This change adds an assertion to check that the sum isn't *too* far from summing to 1

### CODE REVIEW

**Comment by beyarkay** [comment]
The CI build failed with 
```
ERROR: The project you're trying to build requires Bazel 6.1.0 (specified in /workspace/.bazelversion), but it wasn't found in /usr/local/lib/bazel/bin.
```

which doesn't seem related to my change?

**Comment by beyarkay** [comment]
@gbaned what are the next steps here?
